runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -132,6 +135,23 @@ public void onGameObjectDespawned(GameObjectDespawned event)
 		}
 	}
 
+	@Subscribe
+	public void onGameObjectSpawned(GameObjectSpawned event)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+		GameObject object = event.getGameObject();
+
+		// If the Lovakite ore respawns before the timer is up, remove it
+		if (object.getId() == ROCKS_28596 || object.getId() == ROCKS_28597)
```

Comment:
```suggestion
		if (Rock.getRock(object.getId()) == Rock.LOVAKITE)
```

Reply:
Alternatively, refactor this method and `onWallObjectSpawned` to use a common helper method and add these IDs to the `switch` that already exists there, alongside the `ORE_VEIN` cases.

Reply:
i took your refactor suggestion, that seems more extendable in the future if other rocks end up behaving that way